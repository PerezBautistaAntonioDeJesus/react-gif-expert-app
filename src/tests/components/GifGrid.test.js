import { shallow } from "enzyme";
import { React } from "react";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock('../../hooks/useFetchGifs');


describe('tests for <GidGrid /> component', () => { 

    
    test('should create the snapshot', () => { 
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={''} />);

        expect( wrapper ).toMatchSnapshot();
        
     });


     test('should have items when images have been loaded', () => { 
         const fakeValues = [
            {
                id: '23er23',
                url: 'http://fakeurl.com',
                title: 'fake title'
            },
            {
                id: 'iuu9238',
                url: 'http://fakeurl.com',
                title: 'fake title'
            }
         ]
        useFetchGifs.mockReturnValue({
            data: fakeValues,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={''} />);
        
        expect( wrapper.find( 'p' ).exists() ).toBe(false);

        expect( wrapper.find('GifGridItem').length ).toBe( fakeValues.length )
        
      })
    
 })