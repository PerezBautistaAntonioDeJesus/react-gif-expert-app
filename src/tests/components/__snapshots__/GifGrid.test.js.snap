// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <GidGrid /> component should create the snapshot 1`] = `
<Fragment>
  <h3>
     
  </h3>
  <p>
     loading... 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
